lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeUI.cpp

```diff
@@ -356,7 +366,23 @@ bool ProxyShapeUI::select(MSelectInfo& selectInfo, MSelectionList& selectionList
 
   auto selected = false;
 
-  auto removeVariantFromPath = [] (const SdfPath& path) -> SdfPath
+#if defined(WANT_UFE_BUILD)
+  auto pickUfePathPrim = [proxyShape](const SdfPath& path) -> SdfPath {
+    if (ArchHasEnv("MAYA_WANT_UFE_SELECTION")) {
+      // Get only Xform types for ufe selection
+      UsdPrim prim = proxyShape->getUsdStage()->GetPrimAtPath(path);
+      while (prim && prim.GetTypeName() != "Xform")
```

Comment:
Is there a way to do this test with a TfToken for "Xform"? (It should be a little quicker than the strcmps used here)

Reply:
Agreed. Change done.